### PR TITLE
[Feature/ASV-1746] Seasonal Promotions server side

### PR DESCRIPTION
**Pull request by jdandrade**
**What does this PR do?**

   Goal of this is to not have any reference to seasonal promotion ID's, titles and background - dependencies we had in order to create new promotions server side. Now we should be able to create seasonal promotions server side without changing anything in the app.
   To not reference any ID, we had to make a request to a new webservice that gives us promotion id's given a promotion type.
   Views affected are both Home and Promotion view.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] PromotionsManager.java
- [ ] Home.java
- [ ] PromotionsFragment.java 

**How should this be manually tested?**

  Home promotion dialog and Promotions view. Charles rewrite included in comments -> [ASV-1746](https://aptoide.atlassian.net/browse/ASV-1746)

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1746](https://aptoide.atlassian.net/browse/ASV-1746)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/promotions/PromotionsManager.java [hunk 1]
```diff
@@ -62,10 +63,18 @@ public PromotionsManager(PromotionViewAppMapper promotionViewAppMapper,
...
-        .map(
-            appsList -> new PromotionsModel(appsList, getTotalAppc(appsList), isWalletInstalled()));
+  public Single<PromotionsModel> getPromotionsModel(String promotionsType) {
+    return promotionsService.getPromotions(promotionsType)
+        .flatMap(promotions -> {
+          if (promotions.isEmpty()) {
+            return Single.just(new ArrayList<PromotionApp>())
+                .map(__ -> PromotionsModel.ofError());
+          }
+          PromotionMeta meta = promotions.get(0);
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/promotions/PromotionsManager.java]: Why do you get the promotion 0 here ? weren't we supposed to send a model with all the promotions ?

File: app/src/main/java/cm/aptoide/pt/promotions/PromotionsService.java [hunk 2]
```diff
@@ -101,6 +105,32 @@ public String getWalletAddress() {
...
 
+  public Single<List<PromotionMeta>> getPromotions(String type) {
+    return GetPromotionsRequest.of(type, bodyInterceptorPoolV7, okHttpClient, converterFactory,
+        tokenInvalidator, sharedPreferences)
+        .observe()
+        .map(promotionsResult())
+        .toSingle();
+  }
+
+  @NonNull private Func1<GetPromotionsResponse, List<PromotionMeta>> promotionsResult() {
```
Comment (phillipsgoncalves) [inline app/src/main/java/cm/aptoide/pt/promotions/PromotionsService.java]: Why not creating a method and injecting the result inside of it like we usually do ?

File: app/src/main/java/cm/aptoide/pt/promotions/PromotionsManager.java (hunk 1 above)
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/promotions/PromotionsManager.java]: Here we are fetching the first promotion from that promotion type. If there are 2 seasonal promotions at the same time, we will show the most recent one.

File: app/src/main/java/cm/aptoide/pt/promotions/PromotionsService.java (hunk 2 above)
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/promotions/PromotionsService.java]: In my opinion it makes the getPromotions method cleaner, but I agree that it should be the same pattern we use everywhere else.

**Pull request merged**